rpmlint/pkg.py

```diff
@@ -759,26 +759,145 @@ def __init__(self, name, is_source=False):
         self.files = {}
         self.ghost_files = {}
 
+        # header is a dictionary to mock rpm metadata
+        self.header = {
+            rpm.RPMTAG_REQUIRENAME: [],
+            rpm.RPMTAG_REQUIREFLAGS: [],
+            rpm.RPMTAG_REQUIREVERSION: [],
+            rpm.RPMTAG_CONFLICTNAME: '',
+            rpm.RPMTAG_CONFLICTFLAGS: '',
+            rpm.RPMTAG_CONFLICTVERSION: '',
+            rpm.RPMTAG_PROVIDENAME: '',
+            rpm.RPMTAG_PROVIDEFLAGS: '',
+            rpm.RPMTAG_PROVIDEVERSION: '',
+            rpm.RPMTAG_OBSOLETENAME: '',
+            rpm.RPMTAG_OBSOLETEFLAGS: '',
+            rpm.RPMTAG_OBSOLETEVERSION: '',
+            rpm.RPMTAG_RECOMMENDNAME: '',
+            rpm.RPMTAG_RECOMMENDFLAGS: '',
+            rpm.RPMTAG_RECOMMENDVERSION: '',
+            rpm.RPMTAG_SUGGESTNAME: '',
+            rpm.RPMTAG_SUGGESTFLAGS: '',
+            rpm.RPMTAG_SUGGESTVERSION: '',
+            rpm.RPMTAG_ENHANCENAME: '',
+            rpm.RPMTAG_ENHANCEFLAGS: '',
+            rpm.RPMTAG_ENHANCEVERSION: '',
+            rpm.RPMTAG_SUPPLEMENTNAME: '',
+            rpm.RPMTAG_SUPPLEMENTFLAGS: '',
+            rpm.RPMTAG_SUPPLEMENTVERSION: '',
+        }
+
+    def _gather_aux(self, header, xs, nametag, flagstag, versiontag,
+                    prereq=None):
+        names = header[nametag]
+        flags = header[flagstag]
+        versions = header[versiontag]
+
+        if versions:
+            for loop in range(len(versions)):
+                name = byte_to_string(names[loop])
+                evr = stringToVersion(byte_to_string(versions[loop]))
+                if prereq is not None and flags[loop] & PREREQ_FLAG:
+                    prereq.append((name, flags[loop] & (~PREREQ_FLAG), evr))
+                else:
+                    xs.append(DepInfo(name, flags[loop], evr))
+        return xs, prereq
+
+    def _gather_dep_info(self):
```

Comment:
The same as above, just move this method to the `AbstractPkg` class, so it's inherited and can be removed from the `Pkg` class too.